Sources/Datadog/DatadogInternal/Context/LaunchTime.swift

```diff
@@ -13,7 +13,9 @@ import Foundation
     ///
     /// If the `UIApplication.didBecomeActiveNotification` has not yet been received by the
     /// time this value is provided, it will represent the time interval between now and the process start time.
-    /* public */ let launchTime: TimeInterval
+    /* public */ let launchTime: TimeInterval?
+
+    /* public */ let launchDate: Date?
```

Comment:
Both interfaces are similar but decorrelated, `LaunchTime` is publicly exposed by `DatadogInternal` while the `__dd_private_AppLaunchHandler` is used internally by the publisher.

`DatadogInternal` allow `nil` for the launch date because it needs to defined a [`.zero`](https://github.com/DataDog/dd-sdk-ios/blob/feature/v2/Sources/Datadog/DatadogInternal/Context/LaunchTime.swift#L24L26) value. So we either make `DatadogContext/launchTime` optional, are we keep `DatadogContext/launchTime/launchDate` optional 🤔 
I think it makes more sense to make `DatadogContext/launchTime` optional, I will see what's the impact of this changes.